### DISCUSSION TITLE
Drop support for end-of-life Linux distributions in Netty 4.2

### DISCUSSION THREAD

**Discussion by slandelle** [netty/netty#14091] [Ideas]
Hey,

Netty currently requires [CentOS 6 compatibility](https://github.com/netty/netty/blob/4.1/.github/workflows/ci-build.yml#L41) for its binary libraries.
We at Gatling had to go to crazy length to make the brotli4j build compatible (cmake, gcc, andglibc versions).

IMO, it doesn't make any sense to keep on supporting distributions that have reached end-of-life.

Regarding CentOS Linux:
* CentOS 6 reached end-of-life on November 30, 2020
* The last supported CentOS release train is CentOS 7 which is reaching end-of-life at the end of this month, see https://endoflife.software/operating-systems/linux/centos
* It's recommend to switch to CentOS Stream 9, see https://blog.centos.org/2023/04/end-dates-are-coming-for-centos-stream-8-and-centos-linux-7/

I think Netty 4.2 is an opportunity to revisit this requirement.

WDYT?

**Comment by chrisvest** [comment]
Jumping to at least 8 should not be a problem. I'm not sure where people are on the long tail of 8 to 9 upgrades.

**Reply by normanmaurer** [reply]
Agree with @chrisvest

**Comment by slandelle** [comment]
It's not my field of expertise, but it seems the images are not properly maintained:
* the [CentOS 6.10 image](https://hub.docker.com/layers/library/centos/centos6.10/images/sha256-ea2905089c8cd9c420ca8b877c3e08ddd732be0fb1d350ae5b87a49e9119a4ff?context=explore) was last pushed 3 years ago and suffers from multiple vulnerabilities
* things are worse for the [CentOS 8 image](https://hub.docker.com/layers/library/centos/centos8/images/sha256-a1801b843b1bfaf77c501e7a6d3f709401a1e0c83863037fa3aab063a7fdb9dc?context=explore)

I'm not sure there's any official Docker image for CentOS Stream.

I wonder if CentOS is a viable solution.

**Comment by linghengqian** [comment]
- I observed the current discussion from https://github.com/oracle/graalvm-reachability-metadata/issues/377 . Another topic is that I have a few instances on the discontinued CentOS Stream 8, which is probably harmless. 
- Docker Hub is no longer trustworthy for the current CentOS Stream. Apparently this is because of pull limits on Docker Hub.
- https://www.centos.org/download/ has marked the location of the CentOS Stream Container. Located at https://quay.io/repository/centos/centos .
    - For CentOS Stream 8, we can get it through `docker pull quay.io/centos/centos:stream8`.
    - For CentOS Stream 9, we can get it through `docker pull quay.io/centos/centos:stream9`.
    - For the unreleased CentOS Stream 10, we can get it through `docker pull quay.io/centos/centos:stream10-development`.
